std-bits/table/src/main/java/org/enso/table/aggregations/Last.java

```diff
@@ -1,27 +1,59 @@
 package org.enso.table.aggregations;
 
 import org.enso.table.data.column.storage.Storage;
+import org.enso.table.data.index.MultiValueKey;
 import org.enso.table.data.table.Column;
 
+import java.util.Arrays;
 import java.util.List;
 
 public class Last extends Aggregator {
   private final Storage storage;
+  private final Storage[] ordering;
   private final boolean ignoreNothing;
 
   public Last(String name, Column column, boolean ignoreNothing) {
+    this(name, column, new Column[0], ignoreNothing);
+  }
+
+  public Last(String name, Column column, Column[] ordering, boolean ignoreNothing) {
     super(name, Storage.Type.OBJECT);
     this.storage = column.getStorage();
+    this.ordering =
+        ordering == null
+            ? new Storage[0]
+            : Arrays.stream(ordering).map(Column::getStorage).toArray(Storage[]::new);
     this.ignoreNothing = ignoreNothing;
   }
 
   @Override
   public Object aggregate(List<Integer> indexes) {
-    for (int i = indexes.size() - 1; i >= 0; i--) {
-      Object value = storage.getItemBoxed(indexes.get(i));
-      if (!ignoreNothing || value != null) {
-        return value;
+    if (ordering.length == 0) {
+      for (int i = indexes.size() - 1; i >= 0; i--) {
+        Object value = storage.getItemBoxed(indexes.get(i));
+        if (!ignoreNothing || value != null) {
+          return value;
+        }
+      }
+    } else {
+      MultiValueKey key = null;
+      Object current = null;
+
+      for (int row : indexes) {
+        Object value = storage.getItemBoxed(row);
+        if (ignoreNothing && value == null) {
+          continue;
+        }
+
+        MultiValueKey newKey =
+            new MultiValueKey(Arrays.stream(this.ordering).map(o -> o.getItemBoxed(row)).toArray());
+        if (key == null || key.compareTo(newKey) <= 0) {
+          key = newKey;
+          current = storage.getItemBoxed(row);
+        }
       }
```

Comment:
Seems to me that if we iterate using `int row : rows`, in case of multiple rows sharing the same ordering key, we may get the same result for `First` and `Last` here. I think we were trying in DB to avoid this, so maybe it would be good to do so here too? I.e. to iterate from the back too so if two rows share the key, a different one is selected than in `First`.

I think we can add a check to the edge case test ensuring that on shared key the results of First and Last are different (if the values are different of course and there's > 1 row obviously).

Reply:
There is already a slight change in the code to ensure Last returns the last item not the first of the set.

Reply:
Ah ok, I haven't noticed that you have `<=` and not `<` here.
Sounds good then, but maybe would be worth adding a comment about it so that it is clear.

Reply:
I've adjusted to be indexes backwards for clarity